lld/ELF/Writer.cpp

```diff
@@ -2394,6 +2394,40 @@ static uint64_t computeFlags(uint64_t flags) {
   return flags;
 }
 
+// Which sections are covered by CHERI PCC bounds.  Currently this includes
+// executable sections, read-only data sections, and GOTs.
+static bool isCheriBoundsSection(const OutputSection *sec) {
+  uint64_t flags = sec->flags;
+
+  // Non-allocatable sections are not mapped into memory.
+  if (!(flags & SHF_ALLOC))
+    return false;
+
+  // Executable sections are fetched via PCC.
+  if (flags & SHF_EXECINSTR)
+    return true;
+
+  // .got is accessed relative to PCC.
+  if (in.got && sec == in.got->getParent())
+    return true;
+  if (in.mipsGot && sec == in.mipsGot->getParent())
+    return true;
+
+  // .got.plt is accessed relative to PCC.
+  if (sec == in.gotPlt->getParent())
```

Comment:
+ igotPlt for completeness? Redundant in reality as it's always in the same OutputSection as gotPlt (except 32-bit Arm which puts it with GotSection) but avoids needing to think about it.